go/ai/generate.go

```diff
@@ -202,8 +203,9 @@ func GenerateWithRequest(ctx context.Context, r *registry.Registry, opts *Genera
 	var output *ModelOutputConfig
 	if opts.Output != nil {
 		output = &ModelOutputConfig{
-			Format: opts.Output.Format,
-			Schema: opts.Output.JsonSchema,
+			Format:      opts.Output.Format,
+			Schema:      opts.Output.JsonSchema,
+			Constrained: opts.Output.JsonSchema != nil,
```

Comment:
Revert this and the `modelFeatures` and in `conformOutput`, on the part that gets added, set `purpose: "output"` on the metadata and check for that in the plugin. Use that in combination with checking for the `ModelOutputConfig.Constrained` flag to check whether you should do constrained gen, and remove the conformance message.